### DISCUSSION TITLE
StrongSwan setup: Charon issue

### DISCUSSION THREAD

**Discussion by knottyg96** [strongswan/strongswan#1502] [Q&A]
Hello, I am setting up StrongSwan on a new system (Debian Linux 11) using the binary package available for that distribution (StrongSwan 5.9.1).  I am trying to follow the host2host-cert configuration posted [here](https://www.strongswan.org/testing/testresults/ikev2/host2host-cert/index.html), and I am encountering some errors with the Charon daemon. When I first used the strongswan-stater service and tried to initiate a connection, I got the following error:

`swanctl --initiate --child host-host 2`
`connecting to 'unix:///var/run/charon.vici' failed: No such file or directory`

When searching for this error, it seems like this indicates that the Charon daemon isn't running.  However, I checked and the daemon is being started by the strongswan-starter service and is running when this error happens.  When I halt the strongswan-starter service, start the Charon daemon directly using the /usr/lib/ipsec/charon executable, and try to initiate the connection, I then get this error:

`swanctl --initiate --child host-host 2`
`initiate failed: CHILD_SA config 'host-host' not found`

I also get the following output from the Charon process:

`11[CFG] vici initiate CHILD_SA 'host-host'`

I killed the Charon daemon and edited my strongswan.conf file to produce a Charon log file for this post; when I restarted the strongswan-starter service and tried to initiate the connection to create a log of the first error, it then gave me the same CHILD_SA error from when I ran Charon directly. Also, a log file was not produced despite my changes.

Does this indicate that Charon/StrongSwan is not properly configured? Or is something else the problem? What should be my next steps for troubleshooting? Attached are my config files for StrongSwan and swanctl.

Thanks.

[strongswan.conf.txt](https://github.com/strongswan/strongswan/files/10369783/strongswan.conf.txt)
[swanctl.conf.txt](https://github.com/strongswan/strongswan/files/10369784/swanctl.conf.txt)

**Comment by tobiasbrunner** [comment]
> When I first used the strongswan-stater service and tried to initiate a connection, I got the following error:

If you want to use `swanctl` and config via `swanctl.conf`, I'd recommend you use the `strongswan` service provided by the `charon-systemd` package and disable the service above (you don't need `starter` or `charon`, just `charon-systemd`).
